### PR TITLE
Replace `setuptools-scm` with bumpversion

### PR DESCRIPTION
**Pull request by messense**
See the discussion in https://github.com/PyO3/setuptools-rust/pull/271.

* patch release: `bumpversion patch`
* minor release: `bumpversion minor`
* major release: `bumpversion major`

### CODE REVIEW

File: setuptools_rust/version.py [hunk 1]
```diff
@@ -0,0 +1,2 @@
+__version__ = version = "1.4.1"
+__version_tuple__ = version_tuple = tuple(__version__.split("."))
```
Comment (messense) [inline setuptools_rust/version.py:2]: TODO: I think it needs to be converted to int.

File: setuptools_rust/version.py [hunk 2]
```diff
@@ -0,0 +1,4 @@
+__version__ = version = "1.4.1"
+__version_tuple__ = version_tuple = tuple(
+    map(lambda x: int(x[1]) if x[0] < 3 else x[1], enumerate(__version__.split(".")))
```
Comment (messense) [inline setuptools_rust/version.py:3]: This `__version_tuple__` is for compatibility with previous `setuptools-scm`, it's kinda ugly, we could remove it in 2.0

File: .bumpversion.cfg [hunk 3]
```diff
@@ -0,0 +1,8 @@
+[bumpversion]
+commit = True
+tag = True
```
Comment (davidhewitt) [inline .bumpversion.cfg:3]: I wonder if we want `tag = False` ? Usually I let Github releases create the tag.

File: setuptools_rust/version.py (hunk 2 above)
Comment (davidhewitt) [reply setuptools_rust/version.py:3]: 👍 let's note it in #259

File: .bumpversion.cfg (hunk 3 above)
Comment (messense) [reply .bumpversion.cfg:3]: OK

**Review by davidhewitt — APPROVED** [review]
Looks good to me, thanks!

**Pull request merged**